internal/pkg/exec/dockerengine.go

```diff
@@ -264,32 +264,38 @@ func ValidatePlatform(platform *string) error {
 	if platform == nil {
 		return nil
 	}
-	if aws.StringValue(platform) != dockerBuildPlatform(LinuxOS, Amd64Arch) {
-		return fmt.Errorf("platform %s is invalid; %s: %s", aws.StringValue(platform), english.PluralWord(len(validPlatforms), "the valid platform is", "valid platforms are"), english.WordSeries(validPlatforms, "and"))
+	for _, validPlatform := range validPlatforms {
+		if aws.StringValue(platform) == validPlatform {
+			return nil
+		}
 	}
-	return nil
+	return fmt.Errorf("platform %s is invalid; %s: %s", aws.StringValue(platform), english.PluralWord(len(validPlatforms), "the valid platform is", "valid platforms are"), english.WordSeries(validPlatforms, "and"))
 }
 
 // ValidateOS checks if the entered string is a Docker-buildable operating system.
 func ValidateOS(os *string) error {
```

Comment:
tiny nit:
```suggestion
func ValidateOS(os string) error {
```

Partly because it's a little ambiguous whether a `nil` pointer is a valid platform 💭 

I personally think it would have been fine if `ValidatePlatform` was a private function defined in `manifest` pkg - in that case, it makes sense to write a helper function that's tailored for `manifest` (hence accepting `*string` instead of `string`). However if it's an exported function, we should make sure the function name and implementation is semantically consistent and disambiguous.

Reply:
I agree that it could have stayed in the manifest package. I'll move it.